schema_editor/app/scripts/auth/auth-service.js

```diff
@@ -103,25 +116,51 @@
             return isNaN(userId) ? -1 : userId;
         };
 
+        /*
+         * Returns true if currently logged in user has write access (admin or analyst)
+         */
+        module.hasWriteAccess = function() {
+            return $cookies.getObject(canWriteCookieString) || false;
+        };
+
         module.logout =  function() {
             setUserId(null);
             $cookies.remove(tokenCookieString, {path: '/'});
-            $rootScope.$broadcast(events.loggedOut);
-            if (cookieTimeout) {
-                $timeout.cancel(cookieTimeout);
-                cookieTimeout = null;
-            }
+            setCanWrite(false).then(function() {
+                $rootScope.$broadcast(events.loggedOut);
+                if (cookieTimeout) {
+                    $timeout.cancel(cookieTimeout);
+                    cookieTimeout = null;
+                }
 
-            // Hit logout openid endpoint after clearing cookies to log out of API session
-            // created by SSO login, too. Refreshes page for token/user cookies to clear as well.
-            // Redirects back to current location when done.
-            $window.location.href = [ASEConfig.api.hostname,
-                '/api-auth/logout/?next=',
-                $window.location.href].join('');
+                // Hit logout openid endpoint after clearing cookies to log out of API session
+                // created by SSO login, too. Refreshes page for token/user cookies to clear as well.
+                // Redirects back to current location when done.
+                $window.location.href = [ASEConfig.api.hostname,
+                    '/api-auth/logout/?next=',
+                    $window.location.href].join('');
+            });
         };
 
         return module;
 
+
+        function setCanWrite(canWrite) {
+            var dfd = $q.defer();
+
+            // set cookie after a timeout. Angular polls every 100ms to see if there are any
+            // cookies to set; necessary to make sure cookie is set before full page refresh.
+            //
+            // https://github.com/angular/angular.js/blob/1bb33cccbe12bda4c397ddabab35ba1df85d5137/src/ng/browser.js#L102
+            // https://github.com/angular/angular.js/blob/1bb33cccbe12bda4c397ddabab35ba1df85d5137/src/ngCookies/cookies.js#L58-L66
+            $timeout(function() {
```

Comment:
`$timeout` returns a promise, so I think it might be possible to do `return $timeout(...)` directly without constructing your own deferred object here.